file.go

```diff
@@ -134,37 +149,20 @@ func ioCompletionProcessor(h syscall.Handle) {
 
 // asyncIo processes the return value from ReadFile or WriteFile, blocking until
 // the operation has actually completed.
-func (f *win32File) asyncIo(c *ioOperation, deadline time.Time, bytes uint32, err error) (int, error) {
+func (f *win32File) asyncIo(c *ioOperation, timeout timeoutChan, bytes uint32, err error) (int, error) {
```

Comment:
I think we should avoid taking the lock unless we actually go async, so I think this will be necessary.